Removes the ability to permastun ppl w/ slipping on banana peels/soap, adds some inertia

TobyThorne opened the pull request:
- Лежачие космонавтики не подскальзываются;
- Подскальзывание на мыле/банане теперь отбрасывает и космонавтика, и мыло/банан в разные стороны. Инерция, хуё-моё.
- Раньше мыло сохло только от правильного использования. Теперь сохнет и от подскальзываний. По умолчанию в мыле 15 единиц клинера, мытьём в раковине можно поднять до 30 единиц, подскальзывание тратит 3 единицы. На сухом мыле покататься нельзя.
- Банановые кожурки не подскальзывают при ходьбе, только при беге.

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

Commit:
Soapy Nerfy

- Can no longer slip while in weakened state;
- Soap slip now applies 4 weakened instead of 3;
- Soap bars have a chance to smudge when being stepped on;
- Smudged soap (a cleanable decal) is also slippery as fuck;

Commit:
Delete pool.png

Ershovdi commented:
В землю занёрфили, сделайте пониже проценты.

ArmenianOnyx commented:
пошел нахуй, лицо набью не смей

Pos13 commented on code/game/objects/effects/decals/Cleanable/misc.dm:
```diff
@@ -130,3 +130,16 @@
+/obj/effect/decal/cleanable/soap_smudge/Crossed(AM as mob|obj)
+	if(istype(AM, /mob/living))
+		var/mob/living/M = AM
+		M.slip("the [src.name]", 4)
```
Я думаю, надо добавить проверку на бег/шаг. На обычном мыле, наверное, можно подскользнуться и шагом, а вот на раздавленной мыльной куче - вряд ли.

TobyThorne replied:
Ты дев, а не геймдизайнер.

Pos13 replied:
Тем не менее, я могу высказать обоснованное сомнение как игрок. Не нужно быть лицензированным поваром, чтобы оценить вкус еды.

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -20,46 +21,50 @@
 	//So this is a workaround. This also makes more sense from an IC standpoint. ~Carn
 	if(user.client && (target in user.client.screen))
-		to_chat(user, "<span class='notice'>You need to take that [target.name] off before cleaning it.</span>")
+		to_chat(user, SPAN("notice", "You need to take that [target.name] off before cleaning it."))
```
```suggestion
		to_chat(user, SPAN_NOTICE("You need to take that [target.name] off before cleaning it."))
```

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -20,46 +21,50 @@
+		to_chat(user, SPAN("notice", "You need to take that [target.name] off before cleaning it."))
 	else if(istype(target,/obj/effect/decal/cleanable/blood))
-		to_chat(user, "<span class='notice'>You scrub \the [target.name] out.</span>")
+		to_chat(user, SPAN("notice", "You scrub \the [target.name] out."))
```
```suggestion
		to_chat(user, SPAN_NOTICE("You scrub \the [target.name] out."))
```

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -20,46 +21,50 @@
 		target.clean_blood() //Blood is a cleanable decal, therefore needs to be accounted for before all cleanable decals.
 	else if(istype(target,/obj/effect/decal/cleanable))
-		to_chat(user, "<span class='notice'>You scrub \the [target.name] out.</span>")
+		to_chat(user, SPAN("notice", "You scrub \the [target.name] out."))
```
```suggestion
		to_chat(user, SPAN_NOTICE("You scrub \the [target.name] out."))
```

TobyThorne replied:
По*кхем*.

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -20,46 +21,50 @@
 		qdel(target)
 	else if(istype(target,/turf))
-		to_chat(user, "<span class='notice'>You scrub \the [target.name] clean.</span>")
+		to_chat(user, SPAN("notice", "You scrub \the [target.name] clean."))
```
```suggestion
		to_chat(user, SPAN_NOTICE("You scrub \the [target.name] clean."))
```

TobyThorne replied:
Пососёшь.

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -20,46 +21,50 @@
 		T.clean(src, user)
 	else if(istype(target,/obj/structure/sink))
-		to_chat(user, "<span class='notice'>You wet \the [src] in the sink.</span>")
+		to_chat(user, SPAN("notice", "You wet \the [src] in the sink."))
```
```suggestion
		to_chat(user, SPAN_NOTICE("You wet \the [src] in the sink."))
```

TobyThorne replied:
Пососёшь.

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -20,46 +21,50 @@
 		wet()
 	else
-		to_chat(user, "<span class='notice'>You clean \the [target.name].</span>")
+		to_chat(user, SPAN("notice", "You clean \the [target.name]."))
```
```suggestion
		to_chat(user, SPAN_NOTICE("You clean \the [target.name]."))
```

TobyThorne replied:
Пососёшь.

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -10,6 +10,7 @@
 	throw_speed = 4
 	throw_range = 20
 	var/key_data
+	var/durability = 50
```
```suggestion
	var/durability = 50 // consider 100 as maximum possible durability
```
Предлагаю оставить небольшой комментарий, чтобы создателям нового мыла было проще брать цифры с потолка для _durability_.

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -20,46 +21,50 @@
 /obj/item/weapon/soap/attack(mob/living/target, mob/living/user, target_zone)
 	if(target && user && ishuman(target) && ishuman(user) && !target.stat && !user.stat && user.zone_sel &&user.zone_sel.selecting == BP_MOUTH)
-		user.visible_message("<span class='danger'>\The [user] washes \the [target]'s mouth out with soap!</span>")
+		user.visible_message(SPAN("danger", "\The [user] washes \the [target]'s mouth out with soap!"))
```
```suggestion
		user.visible_message(SPAN_DANGER("\The [user] washes \the [target]'s mouth out with soap!"))
```

TobyThorne replied:
Пососёшь.

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -20,46 +21,50 @@
 	if(istype(I, /obj/item/weapon/key))
 		if(!key_data)
-			to_chat(user, "<span class='notice'>You imprint \the [I] into \the [src].</span>")
+			to_chat(user, SPAN("notice", "You imprint \the [I] into \the [src]."))
```
```suggestion
			to_chat(user, SPAN_NOTICE("You imprint \the [I] into \the [src]."))
```

Pos13 replied:
Мне лень дальше писать замены _SPAN("warning", _ на _SPAN_WARNING(_, ты и так меня понял.

TobyThorne replied:
Пососёшь.

Pos13 replied:
Хрюкнул. Теперь иди исправлять.

TobyThorne replied:
Громче. Если хочешь - сам меняй нормальные макросы на эту нечитаемую капсовую хуиту, после того как примут (если, да).

Pos13 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -20,46 +21,50 @@
+		if(M.slip("the [src.name]", 4))
+			if(prob(100-durability))
+				new /obj/effect/decal/cleanable/soap_smudge(loc)
+				visible_message(SPAN("danger", "[M] smudges \the [src]!"))
```
```suggestion
				M.visible_message(SPAN_WARNING("[M] smudges \the [src]!"), SPAN_WARNING("You smudge \the [src]!"))
```
Думаю, SPAN_DANGER - это сообщение уровня непосредственной опасности, вроде пистолетного выстрела.

TobyThorne committed:
Update code/game/objects/items/weapons/soap.dm

Co-authored-by: Pos13 <64479446+Pos13@users.noreply.github.com>

Commit:
Merge branch 'dev' of https://github.com/ChaoticOnyx/OnyxBay into soapnerf

Commit:
banananerf

- spessmen no longer slip on banana peels while walking

TobyThorne commented:
* На банановой кожуре теперь нельзя подскользнуться, если мув интент - шаг, а не бег. 
* Пофиксил конфликты.

Commit:
Update soap.dm

lowered chances for soap to be smudged

TobyThorne commented:
Обновлены шансы раздавливания:
Обычное мыло: 35%
НТ-мыло: 35%
Секурити-мыло: 35%
Элитное мыло (aka кэпское): 15%
Синдимыло: 25%
ЗоЛоТоЕ мЫлО: 5%

Voldirs commented:
![image](https://user-images.githubusercontent.com/4889771/95295711-42885880-0880-11eb-9b6f-16eca0fc905d.png)

Epicus7 commented on code/game/objects/effects/decals/Cleanable/misc.dm:
```diff
@@ -130,3 +130,16 @@
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "soap"
+
+/obj/effect/decal/cleanable/soap_smudge/Crossed(AM as mob|obj)
```
```suggestion
/obj/effect/decal/cleanable/soap_smudge/Crossed(atom/movable/AM)
```

Epicus7 commented on code/game/objects/items/weapons/soap.dm:
```diff
@@ -10,6 +10,7 @@
 	throw_speed = 4
 	throw_range = 20
 	var/key_data
+	var/durability = 65 // consider 100 as the maximum possible durability
```
```suggestion
	var/smash_chance = 65
```

Вот так лучше назвать переменную и не нужен будет стремный коммент.

PlinHost commented:
У нас нет вообще никакого способа получения мыла (кроме синдимыла), с такими шансами все мыло на станции тупо закончится.
Я считаю, достаточно убрать стан по лежачим. Кожуру легко подобрать, так что нет смысла убирать подскальзывание на шаге.

TobyThorne commented:
> У нас нет вообще никакого способа получения мыла (кроме синдимыла), с такими шансами все мыло на станции тупо закончится.
> Я считаю, достаточно убрать стан по лежачим. Кожуру легко подобрать, так что нет смысла убирать подскальзывание на шаге.

Так пусть заканчивается, не? Если использовать по назначению - то не закончится, если использовать его в боёвке - то логично, что это должен быть расходник, а не батон на максималках.

Стан по лежачим и так убрал, но это не работает от слова совсем. Космонавтик встаёт не мгновенно, даже когда weakened уже равно нолю. Вот этого времени достаточно чтобы всё равно с помощью пулла вечно держать в стане.

MrMagick2104 commented:
> Так пусть заканчивается, не?

Проблема в том, что восполнить запасы сложно.
Нужно тогда добавить в карго ящик с мылом, на брусков пять.

Commit:
poopdate

- Generic and NT soap smudge chance increased from 35% to 50%;
- Added a Five Soap Bars crate to cargo supplies list;

TobyThorne commented:
- Шанс раздавливания обычного и НТшного мыла снова увеличен до 50%;
- В карго добавлен ящик с пятью брусками мыла за 15 очков;

Epicus7 commented:
Короче хороший ПР, мне нравится. По геймдизу тоже одобрил.

Epicus7 commented:
Алсо этот ПР закрывает #2117

MorJams commented:
раздавливание не нужно, со станбатоном и так достаточно различий, мыло - столб классической боёвки, лучше как Плин сказал делайте

поменьше стана, убирание станлока во время тасканий и всё

Commit:
slips reworked

Redone the whole PR. New state:
- Slipping on soap and banana peels now throws both the human and the object flying in opposite directions;
- Soap gets drier after getting slipped on;
- Humans can't slip on a banana peel while walking;

Commit:
Update carbon.dm

Destrolaric commented:
> Полностью всё переделал. Всё вышеописанное можно забыть, что есть теперь:
> 
> * Лежачие космонавтики не подскальзываются (проблема та же, что и раньше - подскользнуться можно непосредственно во время вставания);
> * Подскальзывание на мыле/банане теперь отбрасывает и космонавтика, и мыло/банан в разные стороны. Инерция, хуё-моё.
> * Раньше мыло сохло только от правильного использования. Теперь сохнет и от подскальзываний. По умолчанию в мыле 15 единиц клинера, мытьём в раковине можно поднять до 30 единиц, подскальзывание тратит 3 единицы. На сухом мыле покататься нельзя.
> * Банановые кожурки не подскальзывают при ходьбе, только при беге.
Подскальзывание в стену тестил?

TobyThorne committed:
Update carbon.dm

Commit:
extra argument removal
